src/aggregated_range_proof/dealer.rs

```diff
@@ -0,0 +1,174 @@
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::traits::Identity;
+use generators::GeneratorsView;
+use inner_product_proof;
+use proof_transcript::ProofTranscript;
+use std::clone::Clone;
+use util;
+
+use super::messages::*;
+
+/// Dealer is an entry-point API for setting up a dealer
+pub struct Dealer {}
+
+impl Dealer {
+    /// Creates a new dealer with the given parties and a number of bits
+    pub fn new(
+        transcript: &mut ProofTranscript,
+        n: usize,
+        m: usize,
+    ) -> Result<DealerAwaitingValues, &'static str> {
+        transcript.commit_u64(n as u64);
+        transcript.commit_u64(m as u64);
+        Ok(DealerAwaitingValues { n, m })
+    }
+}
+
+/// When the dealer is initialized, it only knows the size of the set.
+pub struct DealerAwaitingValues {
+    n: usize,
+    m: usize,
+}
+
+impl DealerAwaitingValues {
+    /// Combines commitments and computes challenge variables.
+    pub fn receive_value_commitments(
+        self,
+        transcript: &mut ProofTranscript,
+        vc: &Vec<ValueCommitment>,
+    ) -> (DealerAwaitingPoly, ValueChallenge) {
+        // TODO: test that vc is length `m`.
```

Comment:
I think for now it would be OK to just `assert!` this (causing a panic if it fails) for now and deal with the error handling in #57